### ISSUE TITLE
GFM links containing link in label are rendering wrong

### ISSUE THREAD

**Issue by nsychev** [jgm/pandoc#10333]
**Explain the problem.**

Hi! Thanks for your awesome tool.

I'm trying to convert this link using GFM reader:

```markdown
[https://example.org](https://example.org)
```

For example, we can use this command:

```
echo '[https://example.org](https://example.org)' | pandoc -f gfm -t html
```

I expect output to be the same as in `-f markdown`:

```html
<p><a href="https://example.org">https://example.org</a></p>
```

but instead I get this:

```
<p>[<a
href="https://example.org](https://example.org)">https://example.org](https://example.org)</a></p>
```

I think pandoc tries to detect links inside square brackets `[...]` but shouldn't (when I have `https://example.` it still works fine, but `https://example.o` is broken).

**Pandoc version?**

I have pandoc 3.3 locally and reproduced it at https://pandoc.org/try/

**Comment by nsychev** [comment]
It should be obvious, but the problem is in (enabled by default) extension `autolink_bare_uris`, should I disable it, everything works fine. But I think it should work even with it being turned on.

**Comment by jgm** [comment]
Yes, agreed.  This would need to be changed in jgm/commonmark-hs.
And in fact there is already an issue:
https://github.com/jgm/commonmark-hs/issues/156

**Comment by jgm** [comment]
I'll keep this open as a reminder.  It would be good to fix this.

**Comment by tarleb** [comment]
I tried this just now and couldn't reproduce it with the recent pandoc version. Has this already been fixed in commonmark-hs?

**Comment by jgm** [comment]
No.
```
% pandoc -f gfm -t native
[https://example.org](https://example.org)
[ Para
    [ Link
        ( "" , [] , [] )
        [ Link
            ( "" , [] , [] )
            [ Str "https://example.org" ]
            ( "https://example.org" , "" )
        ]
        ( "https://example.org" , "" )
    ]
]
```
It looks like something might have been done in the HTML writer to address the issue there.
Commit bcf06e7f5e508f548f7060ba3e7281766aa2c1ea
Issue #7585
